Scripts/BuildPhases/LintAppLocalizedStringsUsage.swift

```diff
@@ -0,0 +1,324 @@
+    ///   and storing which parent object they belong to.
+    /// - Used by the `resolveURL` method to find the real path of a `PBXReference`, as we need to navigate from the `PBXReference` object
+    ///   up into the chain of parent `PBXGroup` containers to construct the successive relative paths of groups using `sourceTree = "<group>"`
+    private lazy var referrers: [ObjectUUID: ObjectUUID] = {
+        var referrers: [ObjectUUID: ObjectUUID] = [:]
+        func recurseIfGroup(objectID: ObjectUUID, indent: String = "") {
+            guard let group = try? (self.pbxproj.object(id: objectID) as PBXGroup) else { return }
+            for childID in group.children {
+                referrers[childID] = objectID
+                recurseIfGroup(objectID: childID, indent: "\(indent)  ")
+            }
+        }
+        recurseIfGroup(objectID: self.pbxproj.rootProject.mainGroup)
+        return referrers
+    }()
+}
+
+// Convenience methods and properties
+extension Xcodeproj {
+    /// Builds an `Xcodeproj` instance by parsing the an `.xcodeproj` or `pbxproj` file at the provided path
+    convenience init(path: String) throws {
+        try self.init(url: URL(fileURLWithPath: path))
+    }
+
+    /// The directory where the `.xcodeproj` resides. 
+    var projectDirectory: URL { projectURL.deletingLastPathComponent().deletingLastPathComponent() }
+    /// The list of `PBXNativeTarget` targets in the project. Convenience getter for `PBXProjFile.nativeTargets`
+    var nativeTargets: [PBXNativeTarget] { pbxproj.nativeTargets }
+    /// The list of `PBXBuildFile` files a given `PBXNativeTarget` will build. Convenience getter for `PBXProjFile.buildFiles(for:)`
+    func buildFiles(for target: PBXNativeTarget) -> [PBXBuildFile] { pbxproj.buildFiles(for: target) }
+
+    /// Finds the full path / URL of a `PBXBuildFile` based on the groups it belongs to and their `sourceTree` attribute
+    func resolveURL(to buildFile: PBXBuildFile) -> URL {
+        do {
+            let fileRef = try self.pbxproj.object(id: buildFile.fileRef) as PBXFileReference
+            return resolveURL(objectUUID: buildFile.fileRef, object: fileRef)
+        } catch {
+            // Cover `XCVersionGroup` (like `*.xcdatamodel`) and `PBXVariantGroup` (like `*.strings`)
+            let fileRef = try! self.pbxproj.object(id: buildFile.fileRef) as PBXGroup
+            return resolveURL(objectUUID: buildFile.fileRef, object: fileRef)
+        }
+    }
+
+    /// Finds the full path / URL of a PBXReference (`PBXFileReference` of `PBXGroup`) based on the groups it belongs to and their `sourceTree` attribute
+    private func resolveURL<T: PBXReference>(objectUUID: ObjectUUID, object: T) -> URL {
+        if objectUUID == self.pbxproj.rootProject.mainGroup { return URL(fileURLWithPath: ".", relativeTo: projectDirectory) }
+
+        switch object.sourceTree {
+        case .absolute:
+            return URL(fileURLWithPath: object.path!)
```

Comment:
A `PBXReference` with `sourceTree = absolute` but no `path` would make no sense, hence the force unwrap.

Should I catch potential pbxproj inconsistencies like that and make it throw instead of `!`? Or maybe at least a `fatalError(message)`…